### DISCUSSION TITLE
Next-Intl and Auth0 - Need some help to create middleware.ts

### DISCUSSION THREAD

**Discussion by FabioDainese** [amannn/next-intl#848] [Q&A]
Hi guys 👋 In my NextJS v14 project I would like to use both *Next-Intl* and *Auth0*, but I’m not quite sure on how to configure the `middleware.ts` file, since both libraries use it.

Basically, here’s the code that *next-intl* needs to export from the `middleware.ts` file:
```javascript
import createMiddleware from "next-intl/middleware";
import { locales, localePrefix, pathnames } from "@/navigation";

export default createMiddleware({
	defaultLocale: "en",
	localePrefix,
	locales,
	pathnames,
});

export const config = {
	// Skip all paths that should not be internationalized. This example skips the
	// folders "api", "_next" and all files with an extension (e.g. favicon.ico)
	matcher: ["/((?!api|_next|.*\\..*).*)"],
};
```

Meanwhile, here’s the code that *auth0* needs to export from the `middleware.ts` file ([doc](https://developer.auth0.com/resources/guides/web-app/nextjs/basic-authentication#use-middleware-to-protect-next-js-pages)):
```javascript

export default withMiddlewareAuthRequired();

export const config = {
    // Secure the following pages
    matcher: ["/profile"],
};
```
Any idea on how to properly combine the two packages?

Finally, here’s a [sandbox link](https://codesandbox.io/p/devbox/auth0-next-intl-z4r5ks?file=%2Fmiddleware.ts) of a project with minimal setup in order to play around (N.B. if you fork/download it, keep in mind to also create the `.env.local` file following the `.env.example` boilerplate file).

**Comment by MakersAll8** [comment]
I stumbled upon this a year later trying to get Next 15, auth0 middleware, and next-intl middleware to play nice with each other. Code below worked for me, wondering what worked for you.

```typescript
import createMiddleware from "next-intl/middleware";
import { routing } from "./i18n/routing";
import { auth0 } from "./lib/auth0";
import { NextRequest } from "next/server";

export async function middleware(request: NextRequest) {
  const auth0Response = await auth0.middleware(request);
  if (auth0Response && request.nextUrl.pathname !== "/") {
    return auth0Response;
  }

  return createMiddleware(routing)(request);
}

export const config = {
  matcher: [
    "/",

    "/(zh|en|ar)/:path*",

    "/((?!_next/image|_next/static|images|api|favicon.ico|_vercel|.*\\..*).*)",
  ],
};

```
